src/main/scala/com/advancedtelematic/campaigner/http/CampaignResource.scala

```diff
@@ -37,6 +38,23 @@ class CampaignResource(extractAuth: Directive1[AuthedNamespaceScope],
     } yield campaignId
   }
 
+  /**
+    * Create and immediately launch a retry-campaign for all the devices that were processed by `mainCampaign` and
+    * failed with a code `failureCode`. We assume there is at least one failed device. Hence, if there were not any
+    * such devices in `mainCampaign`, throw an error.
+    */
+  def retryFailedDevices(ns: Namespace, mainCampaign: Campaign, request: RetryFailedDevices): Future[CampaignId] =
```

Comment:
I would suggest to replace this method and `CreateRetryCampaign` case-class with a `Campaigns.retryCampagn(ns, mainCampaign, errorCode)` method.

Reply:
Sure :) Updated as advised.